plugins/lsp/schemastore.nix

```diff
@@ -0,0 +1,151 @@
+{
+  lib,
+  helpers,
+  config,
+  pkgs,
+  ...
+}:
+with lib;
+helpers.vim-plugin.mkVimPlugin config {
+  name = "schemastore";
+  originalName = "SchemaStore.nvim";
+  defaultPackage = pkgs.vimPlugins.SchemaStore-nvim;
+
+  maintainers = [ maintainers.GaetanLepage ];
+
+  extraOptions =
+    let
+      schemaOpts =
+        let
+          schemaEntry = types.submodule {
+            options = {
+              name = mkOption {
+                type = types.str;
+                description = "The name of the schema.";
+              };
+
+              description = helpers.mkNullOrStr "A description for this schema.";
+
+              fileMatch = helpers.mkNullOrOption (with types; either str (listOf str)) ''
+                Which filename to match against for this schema.
+              '';
+
+              url = mkOption {
+                type = types.str;
+                description = "The URL of this schema.";
+              };
```

Comment:
Same as feedback as `name`: use `mkNullOrStr` or `maybeRaw`

Reply:
I use `mkOption` here to make this option mandatory.
I have changed the type to `maybeRaw str` though.